pool/connection_pool.go

```diff
@@ -1087,17 +1079,17 @@ func (p *ConnectionPool) addConnection(addr string,
 		for _, watcher := range watched {
 			watcher.unwatch(conn)
 		}
-		log.Printf("tarantool: failed initialize watchers for %s: %s", addr, err)
+		log.Printf("tarantool: failed initialize watchers for %s: %s", id, err)
```

Comment:
I think we should ensure that the connection supports watchers and try to initialize them only in the case. Also we should just skip the step.